driver-core/src/main/com/mongodb/MongoWriteException.java

```diff
@@ -33,7 +33,7 @@
      * @param serverAddress the server address
      */
     public MongoWriteException(final WriteError error, final ServerAddress serverAddress) {
-        super(error.getCode(), error.getMessage(), serverAddress);
+        super("Write operation error on server " + serverAddress + ". Write error: " + error + ". ", serverAddress);
```

Comment:
Ensures that the whole write error, including details, will show up as part of the exception message.  This matches the behavior of MongoBulkWriteException.